src/librustdoc/html/render/mod.rs

```diff
@@ -361,7 +280,7 @@ impl Serialize for Generic {
 
 /// Full type of functions/methods in the search index.
 #[derive(Debug)]
-struct IndexItemFunctionType {
+pub struct IndexItemFunctionType {
```

Comment:
```suggestion
pub(crate) struct IndexItemFunctionType {
```

Reply:
This one gets exposed by the cache so it's got to stay public (until we can move search index related stuff to `html::render::SharedContext`)